### PR TITLE
Phase 2

### PR DESCRIPTION
**Pull request by adorjan**

### CODE REVIEW

File: pancake-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			<groupId>org.testng</groupId>
+			<artifactId>testng</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-all</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<!-- Modules -->
```
Comment (BotondSoltesz) [inline pancake-service/pom.xml]: I suggest to move Test dependencies after Modules.

File: pancake-webapp/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-all</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<!-- Modules -->
+		<dependency>
+			<groupId>io.github.pancake</groupId>
+			<artifactId>pancake-service</artifactId>
+		</dependency>
+		<!-- Modules -->
```
Comment (BotondSoltesz) [inline pancake-webapp/pom.xml]: has this comment additional information?

File: pancake-webapp/src/main/java/io/github/pancake/ServletDemo.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        List<Pancake> pancakes = getOrderablePancakes();
+        PrintWriter out = response.getWriter();
+        out.println("<html>");
+        out.println("<body>");
+        out.println("<h1>Pancake</h1>");
+        for (Pancake pancake : pancakes)
```
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/ServletDemo.java]: Please use {}

File: pancake-webapp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		<dependency>
+			<groupId>ch.qos.logback</groupId>
+			<artifactId>logback-classic</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>javax.servlet-api</artifactId>
+			<scope>provided</scope>
+		</dependency>
+		<!-- Spring -->
```
Comment (BotondSoltesz) [inline pancake-webapp/pom.xml]: As far as I know using Spring is a later step. Can you keep that order please?

File: pancake-webapp/src/test/java/io/github/pancake/consumer/PancakeApplicationTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @Test
+    public void testGetOrderablePancakesShouldReceiveAListOfPancakesViaPancakeServiceWhenInvoked() {
+        // GIVEN
+        underTest.setService(mockPancakeService);
+
+        // WHEN
+        underTest.getOrderablePancakes();
+
+        // THEN
+        Mockito.verify(mockPancakeService, Mockito.only()).getOrderablePancakes();
```
Comment (BotondSoltesz) [inline pancake-webapp/src/test/java/io/github/pancake/consumer/PancakeApplicationTest.java]: Please consider to use static import.

File: pom.xml [hunk 6]
```diff
@@ -13,56 +14,105 @@
 		<jcl.slf4j.version>1.7.19</jcl.slf4j.version>
 		<logback.version>1.1.6</logback.version>
 		<servlet.version>3.1.0</servlet.version>
+		<springframework.version>4.2.5.RELEASE</springframework.version>
 		<mockito.version>1.10.19</mockito.version>
 		<testng.version>6.9.10</testng.version>
+		<io.github.pancake.version>1.0-SNAPSHOT</io.github.pancake.version>
```
Comment (BotondSoltesz) [inline pom.xml]: Using 1.0-SNAPSHOT is a little bit weird.

File: pancake-webapp/pom.xml (hunk 4 above)
Comment (adorjan) [inline pancake-webapp/pom.xml]: We discussed it with @gergelyrozsonits to introduce Spring DI in this step.

File: pancake-webapp/src/main/java/io/github/pancake/configuration/PancakeConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ * Spring configuration class.
+ * 
+ * @author Adorjan Nagy
+ */
+@Configuration
+@ComponentScan(value={"io.github.pancake.consumer"})
+public class PancakeConfiguration {
+    @Bean
+    public PancakeService retrievePancakes() {
+        return new PancakeService();
```
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/configuration/PancakeConfiguration.java]: You can use @Service annotation on PancakeService class so this method is unnecessary.

File: pancake-webapp/src/main/java/io/github/pancake/consumer/PancakeApplication.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import io.github.pancake.persistence.base.Pancake;
+import io.github.pancake.service.PancakeService;
+
+/**
+ * PancakeService consumer class.
+ * 
+ * @author Adorjan Nagy
+ */
+@Component
+public class PancakeApplication {
```
Comment (BotondSoltesz) [inline pancake-webapp/src/main/java/io/github/pancake/consumer/PancakeApplication.java]: Please consider to rename this class to PancakeFacade.

File: pancake-service/src/main/java/io/github/pancake/facade/PancakeFacade.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import io.github.pancake.service.PancakeService;
+
+/**
+ * PancakeService consumer class.
+ * 
+ * @author Adorjan Nagy
+ */
+@Component
+public class PancakeFacade {
+    private PancakeService pancakeService;
```
Comment (kicsikrumpli) [inline pancake-service/src/main/java/io/github/pancake/facade/PancakeFacade.java]: might as well be final

File: pancake-service/src/main/java/io/github/pancake/factory/PancakeFactory.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * @author Adorjan Nagy
+ */
+@Component
+public class PancakeFactory implements FactoryBean<List<Pancake>> {
+    private List<Pancake> pancakes;
+
+    @Override
+    public List<Pancake> getObject() {
+        if (pancakes == null) {
```
Comment (kicsikrumpli) [inline pancake-service/src/main/java/io/github/pancake/factory/PancakeFactory.java]: - Is lazy init really necessary?
- leaves the object in an invalid state until the first get :(
- the name `init` already implies this is a constructor like thing - why not put this in a constructor?
- I'd suggest returning a copy of pancakes and not exposing the internal representation via the `pancakes` reference
  - exposes the internals
  - couples implementation and interface

File: pancake-webapp/src/main/java/io/github/pancake/service/configuration/PancakeServiceConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+/**
+ * Spring configuration class.
+ * 
+ * @author Adorjan Nagy
+ */
+@Configuration
+@ComponentScan(value={"io.github.pancake"})
```
Comment (kicsikrumpli) [inline pancake-webapp/src/main/java/io/github/pancake/service/configuration/PancakeServiceConfiguration.java]: - single element arrays can be passed without the curly braces
- `@ComponentScan` allows to pass a class whose package is treated as base class
  - less error-prone
  - more type-safe than a string
  - refactoring might leave string unchanged

File: pancake-webapp/src/main/webapp/WEB-INF/web.xml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+	version="3.0">
+	<servlet>
+		<servlet-name>pancake-app</servlet-name>
+		<servlet-class>io.github.pancake.ServletDemo</servlet-class>
+	</servlet>
+	<servlet-mapping>
+		<servlet-name>pancake-app</servlet-name>
+		<url-pattern>/</url-pattern>
+	</servlet-mapping>
+</web-app>
```
Comment (kicsikrumpli) [inline pancake-webapp/src/main/webapp/WEB-INF/web.xml]: missing new line at the end of the file. It's good form, is worth getting in the habit of

File: pancake-webapp/src/test/java/io/github/pancake/ServletDemoTest.java [hunk 13]
```diff
@@ -27,19 +35,20 @@
...
         // GIVEN
-        Mockito.when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
+        when(mockResponse.getWriter()).thenReturn(mockPrintWriter);
 
         // WHEN
         underTest.doGet(mockRequest, mockResponse);
 
         // THEN
-        Mockito.verify(mockPrintWriter, Mockito.atLeastOnce()).println(Mockito.anyString());
+        verify(mockPrintWriter, atLeastOnce()).println(Mockito.anyString());
```
Comment (kicsikrumpli) [inline pancake-webapp/src/test/java/io/github/pancake/ServletDemoTest.java]: atLeastOnce() is the default, adds unnecessary verbosity

**Pull request merged**